libs/node/src/services/node-crypto-function.service.ts

```diff
@@ -172,24 +173,33 @@ export class NodeCryptoFunctionService implements CryptoFunctionService {
     mac: string | null,
     key: SymmetricCryptoKey,
   ): CbcDecryptParameters<Uint8Array> {
-    const p = {} as CbcDecryptParameters<Uint8Array>;
-    p.encKey = key.encKey;
-    p.data = Utils.fromB64ToArray(data);
-    p.iv = Utils.fromB64ToArray(iv);
-
-    const macData = new Uint8Array(p.iv.byteLength + p.data.byteLength);
-    macData.set(new Uint8Array(p.iv), 0);
-    macData.set(new Uint8Array(p.data), p.iv.byteLength);
-    p.macData = macData;
-
-    if (key.macKey != null) {
-      p.macKey = key.macKey;
-    }
-    if (mac != null) {
-      p.mac = Utils.fromB64ToArray(mac);
+    const dataBytes = Utils.fromB64ToArray(data);
+    const ivBytes = Utils.fromB64ToArray(iv);
+    const macBytes = mac != null ? Utils.fromB64ToArray(mac) : null;
+
+    const innerKey = key.inner();
+
+    if (innerKey.type === EncryptionType.AesCbc256_B64) {
+      return {
+        iv: ivBytes,
+        data: dataBytes,
+        encKey: innerKey.encryptionKey,
+      } as CbcDecryptParameters<Uint8Array>;
+    } else if (innerKey.type === EncryptionType.AesCbc256_HmacSha256_B64) {
+      const macData = new Uint8Array(ivBytes.length + dataBytes.length);
+      macData.set(ivBytes, 0);
+      macData.set(dataBytes, ivBytes.length);
```

Comment:
```suggestion
      const macData = new Uint8Array(ivBytes.byteLength + dataBytes.byteLength);
      macData.set(ivBytes, 0);
      macData.set(dataBytes, ivBytes.byteLength);
```

💭  Should we be using `byteLength` instead of `length` to preserve previous behavior?

Reply:
Not sure it matters since for uint8array, byteLength and length should always be equal, but happy to apply this.